Amplify/Categories/DataStore/Model/Collection/List+Model.swift

```diff
@@ -91,48 +129,43 @@ public class List<ModelType: Model>: Collection, Codable, ExpressibleByArrayLite
         return 0
     }
 
+    /// `limit` is currently not a supported API.
+    @available(*, deprecated, message: "Not supported.")
     public func limit(_ limit: Int) -> Self {
         // TODO handle query with limit
-        self.limit = limit
-        state = .pending
+        loadedState = .notLoaded
```

Comment:
This method shouldn't have any side effects if it's not supported. This implementation would allow someone to call `limit` and reset the loaded state of the list, thus potentially working around some of our state checks elsewhere.

Reply:
my original thinking was to keep the original implementation since that would be the "backwards compatible" thing to do. But the previous implementation was not complete, and since we are marking this as not supported and the internal state management has changed drastically in this PR, it will be safer to remove the implementation in `limit` all together.